File: src/main/java/me/leodev/trollingfreedomreborn/trolls/chat/Nick.java
package me.leodev.trollingfreedomreborn.trolls.chat;

import com.earth2me.essentials.Essentials;
import com.earth2me.essentials.User;
import me.leodev.trollingfreedomreborn.main.Core;
import me.leodev.trollingfreedomreborn.other.Mode;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;

import java.util.ArrayList;

public class Nick implements Listener {

    public static ArrayList<String> Nick1 = new ArrayList();

    public void NickName(final Player p) {
        String p2 = p.getName();
        Essentials ess = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
        User user = ess.getUserMap().getUser(p.getName());
        Nick1.add(p.getName());
        int id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, new Runnable() {
            public void run() {
                String randomString1 = Mode.getString(8, Mode.ALPHANUMERIC);
                //Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(), "nick " + p.getName() + " " + randomString1);
                user.setNickname(randomString1);
            }
        }, 10L, 10L);
        Core.instance.addTask(p, "nick", id);
    }

    public void UnNick(Player p) {
        Essentials ess = (Essentials) Bukkit.getServer().getPluginManager().getPlugin("Essentials");
        User user = ess.getUserMap().getUser(p.getName());

        // String nick = user.getNick();
        if (Nick1.contains(p.getName())) {
            Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(), "nick " + p.getName() + " off");
            Nick1.remove(p.getName());
        } else {
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/chat/Nick.kt
package com.leomadrassi.trollingfreedomreborn.trolls.chat

import com.earth2me.essentials.Essentials
import com.earth2me.essentials.User
import com.leomadrassi.trollingfreedomreborn.main.Core
import com.leomadrassi.trollingfreedomreborn.other.Mode
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.Listener

class Nick : Listener {
    companion object {
        val Nick1 = mutableListOf<String>()
    }

    fun NickName(p: Player) {
        val ess = Bukkit.getServer().pluginManager.getPlugin("Essentials") as Essentials
        val user = ess.userMap.getUser(p.name)
        Nick1.add(p.name)
        val id = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, Runnable {
            val randomString1 = Mode.getString(8, Mode.ALPHANUMERIC)
            user.nickname = randomString1
        }, 10L, 10L)
        Core.instance.addTask(p, "nick", id)
    }

    fun UnNick(p: Player) {
        val ess = Bukkit.getServer().pluginManager.getPlugin("Essentials") as Essentials
        val user = ess.userMap.getUser(p.name)
        if (Nick1.contains(p.name)) {
            Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(), "nick ${p.name} off")
            Nick1.remove(p.name)
        }
    }
}
